### PR TITLE
Add support for cracking older BitShares wallets

### PR DESCRIPTION
**Pull request by kholia**
For some reason, the format auto-detection is failing for the output `$dynamic_84$` hash format.

### CODE REVIEW

**Comment by magnumripper** [comment]
That's expected with no tag. Without the tag, using `--bare-always-valid=Y` *should* work (but wouldn't ensure dynamic_84 was the *first* choice) but that doesn't seem to work for some reason.

But this works for me with auto-detection:
```
$dynamic_84$b335d77abf42f6da1b6d0864129fa176c7c499b70c389b273e719473e5f029f54b9b8914e96a6585db2f16c987e6b988e78a0e497bbe8bc88d62df0eb81b7010
```

File: doc/DYNAMIC_COMPILER_FORMATS.md [hunk 1]
```diff
@@ -193,3 +193,28 @@ Session completed
...
+Press 'q' or Ctrl-C to abort, almost any other key for status
+password123      (?)
+openwall         (?)
+```
+
+NOTE: The inbuilt `dynamic_84` format is a bit faster at cracking such hashes. It can be used via the following command,
+
+```
+$ ../run/john --format=dynamic_84 sample-hashes -w=wordlist
+```
```
Comment (magnumripper) [inline doc/DYNAMIC_COMPILER_FORMATS.md]: Why do we even suggest the `-form=dynamic='sha512(sha512_raw($p))'` if dynamic_84 is faster?

Comment (kholia) [reply doc/DYNAMIC_COMPILER_FORMATS.md]: You have a point. I will remove this part.

**Comment by kholia** [comment]
@magnumripper This is super weird.

```
$ cat hashes 
$dynamic_84$f4bf2f7ba63b375dd21325f06b3b857294d88e760d854125459d4f1b82bf576d2de53ab0f52fd96990bd6ee7cee956c4bf133d79b7c1575cc4624ca0d3da7be4
$dynamic_84$b335d77abf42f6da1b6d0864129fa176c7c499b70c389b273e719473e5f029f54b9b8914e96a6585db2f16c987e6b988e78a0e497bbe8bc88d62df0eb81b7010

$ ../run/john hashes
Using default input encoding: UTF-8
No password hashes loaded (see FAQ)
```

Why is auto-detection failing for me?

```
$ ../run/john --format=dynamic_84 hashes -w=wordlist
Using default input encoding: UTF-8
Loaded 2 password hashes with no different salts (dynamic_84 [sha512(sha512_raw($p)) 256/256 AVX2 4x])
Press 'q' or Ctrl-C to abort, almost any other key for status
password123      (?)
openwall         (?)
2g 0:00:00:00 DONE (2018-06-23 16:34) 100.0g/s 100.0p/s 100.0c/s 200.0C/s password123..openwall
```

**Comment by kholia** [comment]
The hash extractor script does output hashes with tags.

Hopefully, this is a PEBKAC case and not a bug ;)

**Comment by magnumripper** [comment]
I copy-pasted your hashes above to test.in, then
```
$ cat test.in
$dynamic_84$f4bf2f7ba63b375dd21325f06b3b857294d88e760d854125459d4f1b82bf576d2de53ab0f52fd96990bd6ee7cee956c4bf133d79b7c1575cc4624ca0d3da7be4
$dynamic_84$b335d77abf42f6da1b6d0864129fa176c7c499b70c389b273e719473e5f029f54b9b8914e96a6585db2f16c987e6b988e78a0e497bbe8bc88d62df0eb81b7010

$ ../run/john test.in -w
Using default input encoding: UTF-8
Loaded 2 password hashes with no different salts (dynamic_84 [sha512(sha512_raw($p)) 256/256 AVX2 4x])
Warning: no OpenMP support for this hash type, consider --fork=8
Press 'q' or Ctrl-C to abort, almost any other key for status
0g 0:00:00:00 DONE (2018-06-23 18:47) 0g/s 354600p/s 354600c/s 709200C/s Hammer..sss
Session completed
```
What if you run `-test -form:dynamic_84` ? Jim may disable the format in some odd cases.

**Comment by kholia** [comment]
Once `--format=dynamic_84` is specified, things (cracking, benchmarking) work just fine.

I have reproduced this on local macOS and on `super` box.

**Pull request merged**